proxy/config/http.go

```diff
@@ -0,0 +1,68 @@
+package config
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/caarlos0/env/v11"
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+const tag = "latest"
+
+type HTTPProxyConfig struct {
+	RegistryURL  string `env:"REGISTRY_URL" envDefault:"localhost:5000"`
+	Authenticate bool   `env:"AUTHENTICATE" envDefault:"false"`
+	Username     string `env:"USERNAME"     envDefault:""`
+	Password     string `env:"PASSWORD"     envDefault:""`
+}
+
+func LoadHTTPConfig(opts env.Options) (*HTTPProxyConfig, error) {
+	config := HTTPProxyConfig{}
```

Comment:
move this to main. environment variables should be loaded from main

Reply:
This is called in the main file but its implemented in the config file. So should I remove implementation from the config file?

Reply:
Yes. Move it to main. Loading env variables should be done on main